src/sprite_store.cpp

```diff
@@ -399,41 +388,39 @@ int TimedAnimation::GetFrame(int time, const bool loop_around) const {
  * Load a frame set block from a RCD file.
  * @param rcd_file RCD file used for loading.
  * @param sprites Map of already loaded sprites.
- * @return Loading was successful.
  */
-bool TimedAnimation::Load(RcdFileReader *rcd_file, [[maybe_unused]] const ImageMap &sprites)
+void TimedAnimation::Load(RcdFileReader *rcd_file, [[maybe_unused]] const ImageMap &sprites)
 {
-	if (rcd_file->version != 1 || rcd_file->size < 4) return false;
+	rcd_file->CheckVersion(1);
+	if (rcd_file->size < 4) rcd_file->Error("Length too short for header");
 
 	this->frames = rcd_file->GetUInt32();
-	if (static_cast<int>(rcd_file->size) != 4 + 8 * this->frames) return false;
+	if (static_cast<int>(rcd_file->size) != 4 + 8 * this->frames) rcd_file->Error("Wrong length");
```

Comment:
Would it make sense to have a `CheckLength` much like you have a `CheckVersion` ?